### PR TITLE
add an exclusivity list to output options

### PR DESCRIPTION
**Pull request by KristofferC**
If this list is given, only symbols in the list will be output.

If this is deemed an ok idea, I can add tests etc.

### CODE REVIEW

**Comment by Gnimuc** [comment]
This seems to be the first step to #344

**Comment by KristofferC** [comment]
So this should also get everything dependent on the symbols in the exclusive list?

**Comment by Gnimuc** [comment]
> If this list is given, only symbols in the list will be output.

If a symbol in this list depends on other symbols that are not in this list, then the generated Julia code can not be run. 

What #344 described is that we automatically create a minimal symbol set from this list to make sure the generated code is runable.

File: gen/generator.toml [hunk 1]
```diff
@@ -58,6 +58,9 @@ output_ignorelist = [
     "LLVM_CLANG_C_EXTERN_C_END"
 ]
 
+# Only output id in the `output_exclusivelist"
+# output_output_exclusivelist = []
+
```
Comment (Gnimuc) [inline gen/generator.toml:63]:
```suggestion
```
I think we can merge this and keep this feature private for now.

**Comment by KristofferC** [comment]
Thanks! I have use for this in a wrapper I am toying with.

**Pull request merged**